modules/tinymce/src/themes/silver/test/ts/browser/editor/core/LineHeightTest.ts

```diff
@@ -0,0 +1,137 @@
+import { Assertions, Chain, GeneralSteps, Log, UiFinder } from '@ephox/agar';
+import { UnitTest } from '@ephox/bedrock-client';
+import { Arr, Optional } from '@ephox/katamari';
+import { Editor as McEditor, TinyApis, TinyUi } from '@ephox/mcagar';
+import { Attribute, SugarElement } from '@ephox/sugar';
+import Editor from 'tinymce/core/api/Editor';
+import Theme from 'tinymce/themes/silver/Theme';
+
+const settings = {
+  base_url: '/project/tinymce/js/tinymce',
+  toolbar: 'line-height'
+};
+
+const sOpenToolbar = (ui: TinyUi) => GeneralSteps.sequence([
+  ui.sClickOnToolbar('Click on line-height toolbar', '[title="Line height"]'),
+  ui.sWaitForUi('Open the line-height toolbar', '[role="menu"]')
+]);
+
+const sCloseToolbar = (ui: TinyUi) => GeneralSteps.sequence([
+  ui.sClickOnToolbar('Click on line-height toolbar', '[title="Line height"]')
+]);
+
+const sOpenMenu = (ui: TinyUi) => GeneralSteps.sequence([
+  ui.sClickOnMenu('Click on Format menu', 'button:contains("Format")'),
+  ui.sWaitForUi('Wait for Format menu', '[role="menu"]'),
+  ui.sClickOnUi('Click on line height submenu', '[role="menu"] [title="Line height"]')
+]);
+
+const sCloseMenu = (ui: TinyUi) => GeneralSteps.sequence([
+  ui.sClickOnMenu('Click on Format menu', 'button:contains("Format")')
+]);
+
+const menuSelector = (mode: 'menu' | 'toolbar') => ({
+  menu: '[role="menu"]~[role="menu"]', // the line-height submenu is always the *second* menu in the sink
+  toolbar: '[role="menu"]' // the toolbar is always the *only* menu in the sink
+})[mode];
+
+const sSelectLineHeight = (ui: TinyUi, mode: 'menu' | 'toolbar', lineHeight: string) => GeneralSteps.sequence([
+  ui.sWaitForUi(`Wait for the ${ mode } to open`, menuSelector(mode)),
+  ui.sClickOnUi(`Select line-height of ${ lineHeight }`, `[role="menuitemcheckbox"][title="${ lineHeight }"]`)
+]);
+
+const sAssertOptions = (ui: TinyUi, mode: 'menu' | 'toolbar', ideal: string[], current: Optional<string>) => Log.chainsAsStep('TBA', 'Ensuring that correct options are available', [
```

Comment:
I thought we said we shouldn't do this? What happened to the logic discussed on slack?